src/main_loop/proof_upgrader.rs

```diff
@@ -434,53 +471,60 @@ impl UpgradeJob {
             info!("Producing gobbler-transaction for a value of {gobbling_fee}");
-            let gobble_receiver = own_wallet_entropy.nth_symmetric_key(0);
-            let receiver_preimage = gobble_receiver.privacy_preimage();
-            let gobble_receiver = SpendingKey::Symmetric(gobble_receiver);
-            let gobble_receiver = gobble_receiver.to_address().expect(
-                "gobble receiver should have a corresponding address because it is a symmetric key",
-            );
-            let gobbler = TransactionDetails::fee_gobbler(
+            let (utxo_notification_method, receiver_preimage) =
+                Self::gobbler_notification_method_with_receiver_preimage(
+                    own_wallet_entropy,
+                    fee_notification_policy,
+                );
+            let receiver_digest = receiver_preimage.hash();
+            let gobbler_transaction_details = TransactionDetails::fee_gobbler(
                 gobbling_fee,
-                own_wallet_entropy.generate_sender_randomness(
-                    current_block_height,
-                    gobble_receiver.privacy_digest(),
-                ),
+                own_wallet_entropy
+                    .generate_sender_randomness(current_block_height, receiver_digest),
                 mutator_set,
                 old_tx_timestamp,
-                // TODO: Consider using `None` here as UTXOs are already
-                // stored as expected UTXOs by wallet.
-                UtxoNotifyMethod::OnChain(gobble_receiver),
+                utxo_notification_method,
             );
-            let expected_utxos = gobbler
-                .tx_outputs
-                .iter()
-                .map(|x| {
-                    ExpectedUtxo::new(
-                        x.utxo(),
-                        x.sender_randomness(),
-                        receiver_preimage,
-                        UtxoNotifier::FeeGobbler,
-                    )
-                })
-                .collect_vec();
-            let gobbler = PrimitiveWitness::from_transaction_details(&gobbler);
-            let gobbler_proof =
-                SingleProof::produce(&gobbler, triton_vm_job_queue, proof_job_options.clone())
-                    .await?;
+
+            let expected_utxos =
+                if matches!(fee_notification_policy, FeeNotificationPolicy::OffChain) {
+                    gobbler_transaction_details
+                        .tx_outputs
+                        .iter()
+                        .map(|x| {
+                            ExpectedUtxo::new(
+                                x.utxo(),
+                                x.sender_randomness(),
+                                receiver_preimage,
+                                UtxoNotifier::FeeGobbler,
+                            )
+                        })
+                        .collect_vec()
+                } else {
+                    vec![]
```

Comment:
Please keep the style in the original code. The original code uses shadowing to reduce the number of variables in scope, such that what you now call `gobbler_transaction_details` was previously shadowed once (what you now call) `gobbler_primitive_witness` was declared.